From ba63b049c02f02479d6f1c72b3e3185f3dbade50 Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Fri, 22 Mar 2024 11:36:23 +0900
Subject: [PATCH 70/91] video: dma-drm: add color format mappping about
 NV12_10LE40

Upstream-Status: Pending

add mapping relationship between GST and DRM NV12_10LE40 to drm NV15
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 gst-libs/gst/video/video-info-dma.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/gst-libs/gst/video/video-info-dma.c b/gst-libs/gst/video/video-info-dma.c
index c2d7d48..5feb51e 100644
--- a/gst-libs/gst/video/video-info-dma.c
+++ b/gst-libs/gst/video/video-info-dma.c
@@ -68,6 +68,13 @@
 #define DRM_FORMAT_NV61       fourcc_code('N', 'V', '6', '1')   /* 2x1 subsampled Cb:Cr plane */
 #define DRM_FORMAT_NV24       fourcc_code('N', 'V', '2', '4')   /* non-subsampled Cr:Cb plane */
 
+/*
+ * 2 plane YCbCr
+ * index 0 = Y plane, [39:0] Y3:Y2:Y1:Y0 little endian
+ * index 1 = Cr:Cb plane, [39:0] Cr1:Cb1:Cr0:Cb0 little endian
+ */
+#define DRM_FORMAT_NV15		fourcc_code('N', 'V', '1', '5') /* 2x2 subsampled Cr:Cb plane */
+
 /*
  * 3 plane YCbCr
  * index 0: Y plane, [7:0] Y
@@ -668,6 +675,7 @@ static const struct FormatMap
   {GST_VIDEO_FORMAT_P010_10LE, DRM_FORMAT_P010},
   {GST_VIDEO_FORMAT_P012_LE, DRM_FORMAT_P012},
   {GST_VIDEO_FORMAT_BGR10A2_LE, DRM_FORMAT_ARGB2101010},
+  {GST_VIDEO_FORMAT_NV12_10LE40, DRM_FORMAT_NV15},
 };
 /* *INDENT-ON* */
 
-- 
2.43.0

